cache/refresher.go

```diff
@@ -157,3 +161,30 @@ func (s Refresher) handleRemoveEnvironmentEvent(ctx context.Context, environment
 	}
 	return nil
 }
+
+// handleAddApiKeyEvent adds apiKeys to the cache as well as update apikey list for environment.
+func (s Refresher) handleAddApiKeyEvent(ctx context.Context, env, apiKey string) error {
+	s.log.Debug("adding apikey entry for env", "environment", env, "apiKey", apiKey)
```

Comment:
Wondering should we be logging the apiKey? I know it'll be hashed but still not sure if we should log it.